### DISCUSSION TITLE
Lively as screensaver only starting from batch

### DISCUSSION THREAD

**Discussion by Cris--** [rocksdanister/lively#1026] [Q&A]
Hi,

i would use Lively as screensaver only but i failed with the batch.

Black screensaver only:
@echo off
C:
cd "C:\Tools\Lively Wallpaper\"
livelycu setwp --file "E:\Cloud\Tools\Lively Wallpaper\wallpapers\gdkyowfb.3tk" --monitor 1
timeout 2
start "" livelycu screensaver --show true
livelycu closewp --monitor 1

The screensaver runs only two seconds with these command:
`start "" /wait livelycu screensaver --show true`
`livelycu screensaver --show true`

The culprit here is the _closewp_ which breaks the screensaver. Normally if i use _start_ from a batch the batch waits till i exit the program. So i think _livelycu.exe_ starts _lively.scr_ only and **exit then**, screensaver runs but next batch command breaks the screensaver then.

It would be much easier (if i'm not on a completely wrong way with this batch) with this style of command:
`livelycu start --screensaver_mode_only --file <path_of_wallpaper> --monitor x`  ;)

**Comment by rocksdanister** [comment]
Currently screensaver is tied to wallpaper - you cannot have them separate so closing the wallpaper will also close the screensaver.

Separate screensaver functionality is planned for future https://github.com/rocksdanister/lively/issues/587

**Reply by Cris--** [reply]
..but screensaver would be running fine if i find a way to pause the batch if lively.scr (or what ever) is running and if screensaver stops the next is to stop the wallpaper also..

**Reply by rocksdanister** [reply]
livelycu.exe is just sending message to lively program to bring the running wallpaper infront.

lively.scr is also sending the same message to lively program to bring the running wallpaper infront.

>livelycu start --screensaver_mode_only --file <path_of_wallpaper> --monitor x

This should be possible once https://github.com/rocksdanister/lively/issues/587 is resolved

**Reply by Cris--** [reply]
Ok, so i can't detect via batch if lively is running in infront/fullscreen mode or not?

**Reply by rocksdanister** [reply]
I don't how that will help in this case.. livelcu is only one way messaging now -> to lively program only.

Its definitely possible to create a better livelycu with two way communication in the future with the current ongoing rewrite so made an issue tracker https://github.com/rocksdanister/lively/issues/1027

**Reply by Cris--** [reply]
Ok understand now. The batch detect that livelycu isn't running anymore (sent already the command to the running Lively app) and jumps to the next batch command. So if i have no variable or a running process id/name which indicates that liveley is infront/fullscreen i can't do anything further via batch and must wait for your magical coding things!
